From 50272c8237dcda9efc564ee77f3d6d1ac61723eb Mon Sep 17 00:00:00 2001
From: GSandeep24 <sandeepg2404@gmail.com>
Date: Sun, 13 Nov 2016 22:22:08 +0530
Subject: [PATCH 21/24] Fix memory leak in USB mass storage function by Azhar

---
 drivers/usb/gadget/android.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 0606565..c2134e7 100755
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1560,6 +1560,10 @@ static int mass_storage_function_init(struct android_usb_function *f,
 
 static void mass_storage_function_cleanup(struct android_usb_function *f)
 {
+
+	struct mass_storage_function_config *config;
+ 	config = f->config;
+ 	fsg_common_put(config->common);
 	kfree(f->config);
 	f->config = NULL;
 }
-- 
2.10.2

